dist/counters.js

```diff
@@ -0,0 +1,582 @@
+"use strict";
+var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
+    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
+    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
+    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+    return c > 3 && r && Object.defineProperty(target, key, r), r;
+};
+var __metadata = (this && this.__metadata) || function (k, v) {
+    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
+};
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * This module contains all the definitions for Counters decorators and the utility functions and definitions thereof
+*/
+const class_validator_1 = require("class-validator");
+const utils_1 = require("./utils");
+const charts_1 = require("./charts");
+const util_1 = require("util");
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+const osu = require("node-os-utils");
+const pidusage = require("pidusage");
+const assert = require("assert");
+const moving_averages_1 = require("moving-averages");
+const simple_statistics_1 = require("simple-statistics");
+const writeFileAsync = util_1.promisify(fs.writeFile);
+const logPrefix = 'adstest:counters';
+exports.ProcessStatisticsUnits = {
+    cpu: '%',
+    memory: 'bytes',
+    ctime: 'ms',
+    elapsed: 'ms',
+    timestamp: 'ms'
+};
+/**
+ * The default values for CountersOptions.
+ */
+exports.DefaultCountersOptions = { collectionInterval: 200, includeMovingAverages: true, dumpToFile: true, dumpToChart: true, outputDirectory: `${process.env.TEMP}` };
+/**
+ * A class with methods that help to implement the counters collection for a test method.
+ */
+class Counters {
+    /**

```

Comment:
This file seems generated from counters.ts. Is adding this intentional?

Reply:
yeah for now, I was checking in generated files from previous iteration itself. I will try to spend some time to figure it I can avoid this.